test/dynamo/test_misc.py

```diff
@@ -2644,6 +2644,19 @@ def fn(x):
         self.assertEqual(res, [np.array([0]), np.array([1]), np.array([2])])
         self.assertEqual(cnts.frame_count, 1)
 
+    @torch._dynamo.config.patch(specialize_float=False, capture_scalar_outputs=True)
+    def test_unspecialized_float_add(self):
+        def fn(x, y):
+            return x + y
+
+        cnt = CompileCounterWithBackend("inductor")
+        fn_opt = torch._dynamo.optimize(cnt)(fn)
+
+        x = torch.arange(3)
+        self.assertEqual(fn(x, 2.0), fn_opt(x, 2.0))
+        self.assertEqual(fn(x, 3.0), fn_opt(x, 3.0))
+        self.assertEqual(cnt.frame_count, 1)
```

Comment:
It's also possible Inductor "accidentally" supported passing a SymFloat into an addition kernel, so it might be good to peek at the IR post pass and assert that it looks like what we accept.

Reply:
Also, this is better in the test/inductor/test_torchinductor_dynamic_shapes.py or something, since you're going all the way to inductor here

Reply:
Meh, I can't think of a good way to get at the post graph transform graph, except maybe something like was done in https://github.com/pytorch/pytorch/pull/136508